### ISSUE TITLE
Change default port

### ISSUE THREAD

**Issue by Skarsnik** [usb2snes/usb2snes#1]
As discussed in https://github.com/Skarsnik/QUsb2snes/issues/35 the 8080 can be problematic
This needs some work on the client side

**Comment by qwertymodo** [comment]
Since picking a port is basically arbitrary aside from the need for interoperability, I would suggest going with port 64213, as that's the one that Crowd Control has settled on as their alternate port for this purpose.

**Comment by tcprescott** [comment]
We also should be going with 64213.  ALTTPR's upcoming multiworld will only work on 64213.

**Comment by Skarsnik** [comment]
8080 is a valid port for a websocket based server. I can't switch to another port because it will make old client not working.

**Comment by qwertymodo** [comment]
Not using 64213 will mean that new clients won't work, since that's what they're switching to.  There's no reason you can't listen on both, if keeping backwards compatibility is that important.

**Comment by Skarsnik** [comment]
All I read is new clients as 'my client' and listening to both port is also a really ugly solution from a technical point of view.  I understand wanting to make it work with shit software like Spyke but it introduce really weird issue.
Also you can already change on what QUsb2Snes listen to in the config file

**Comment by tewtal** [comment]
I don't feel like using a non-standard port and then trying to force other implementations to use it is the right way to about this honestly. I don't see why ALTTPR's Multiworld has to only use this nonstandard port that pretty much no one else is using. Wouldn't the better option there be to implement a fallback in the client that checks the new port first, and then falls back to 8080? That would make it compatible with all clients and still allow people to use the web socket on the new port if 8080 was causing problems.

**Comment by Berserker66** [comment]
49152 to 65535 is the dynamic port range and *shoud* be left to the OS to assign to, not manually. Listen to port 0 and the OS will pick a random one in that range. As such a unique port below 49152 should be picked, as anything above could already be randomly in use and then you get random crashes trying to bind an already bound port.

**Comment by OsteHovel** [comment]
I think we should go for a fixed port as dynamic assigned port means that the user have to input the port in applications to utilize it

**Comment by RebelusQuo** [comment]
https://en.wikipedia.org/wiki/WebSocket

> .., RFC 6455 states that WebSocket "is designed to work over HTTP ports 443 and 80 as well as to support HTTP proxies and intermediaries," thus making it compatible with the HTTP protocol.

8080 is specified as an alternative port for the HTTP protocol. Thus, using this port is completely valid in this context.

**Comment by Skarsnik** [comment]
`My suggestion would be 23042 as an alternative port, that's 0x5A02 which is the name of the SNES CPU :smile:`

From @tewtal on discord, I kinda like this choice, will probably be in the 0.8 version of QUsb2Snes

**Comment by tcprescott** [comment]
I was talking to KatDevsGames on this topic.  He was thinking something along the lines of 36952.  It is still outside the RFC-defined dynamic port range and its above 32768.  He didn't want to use anything below 32768 because apparently some Linux distributions don't like that.

**Comment by qwertymodo** [comment]
I haven't heard of any issues with ports under 32768 on Linux, only with the reserved range under 1024.  According to the iana, anything between 1024-49151 should be valid. https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml

Personally, I like tewtal's idea, but I'm for anything that gets away from 8080. This isn't a server, it's local IPC, and should be treated as such.

**Comment by tewtal** [comment]
Turns out I got the SNES CPU identifier wrong :) It's 5A22, so it'd be port 23074 if we'd want to use that. That seems to be unused as well though so should be no problem.

**Comment by tewtal** [comment]
I've never had any kind of issues using ports below 32768 either, so it seems like an odd concern unless it's a problem with some very specific Linux distribution. In the end though it doesn't really matter too much what we end up using as long as we all can agree to it :)

**Comment by tcprescott** [comment]
I think its settled, 23074 it is then?

I talked to Kat and he seemed amicable to listen on it with his usb2sens service as well.

**Comment by tewtal** [comment]
Great, we should add that as an additional port for qusb2snes to listen to as well as 8080. I'll make sure to update my SMZ3 multiworld implementation to try connecting to 23074 initially and falling back to 8080 if that doesn't work.

**Comment by tcprescott** [comment]
23042 right?

I think it switched to 23074 somehow but I think we wanted 23042.

**Comment by tewtal** [comment]
Yeah I got it wrong the first time, it should be 23074 (which is 0x5a22)

**Comment by Skarsnik** [comment]
Now the issue is for me on how to go with it.
Open both port ? Open only 8080 and fall back on the other ? Have an option
for the users (the worse imo) ?

Le mar. 25 août 2020 à 14:01, Thomas Backmark <notifications@github.com> a
écrit :

**Comment by tewtal** [comment]
I think the best solution would be to open both ports at the same time, and then we start working on moving clients to the new port as much as possible while keeping backwards compatibility. And at some point in the future when most/all clients have moved over we can disable port 8080 by default and only use the new one.

**Comment by qwertymodo** [comment]
I would go with listening on both ports, and possibly adding a warning dialog when a client app connects to 8080 so they know to update.

**Comment by tcprescott** [comment]
Can we just stick with 23042?  This is what I told Kat it would be and he's already updated the CC sd2snes service to support it

**Comment by tcprescott** [comment]
Kat is going to switch to 23074 for the CC sd2snes service.  This is set in stone and he won't be amicable to making future changes.
